openmrs/src/main/java/org/motechproject/openmrs/service/impl/OpenMRSEncounterServiceImpl.java

```diff
@@ -180,12 +180,14 @@ private void validateEncounter(Encounter encounter) {
         List<Observation> updatedObs = new ArrayList<>();
         if (originalObservations != null) {
             for (Observation observation : originalObservations) {
-                String conceptUuid = conceptService.resolveConceptUuidFromConceptName(config.getName(), observation.getConcept().getName().getName());
-                if (CollectionUtils.isNotEmpty(observation.getGroupsMembers())) {
-                    resolveConceptUuidForConceptNames(config, observation.getGroupsMembers());
+                if (observation.getValue() != null && StringUtils.isNotEmpty(observation.getValue().getDisplay())) {
```

Comment:
please move this condition to separate method. It will be more readable what you are doing and why